fix: Reduce API polling for `Document.from_batch_process_operation()`

holtskinner opened the pull request:
Also added `ClientInfo` to Document AI Clients to collect API usage metrics

Fixes #246 🦕

holtskinner committed:
fix: Add polling interval and timeout parameters for `Document.from_batch_process_operation()`

fixes #246

Also added `ClientInfo` to Document AI Clients to collect API usage metrics

holtskinner committed:
Change default polling interval to `0`.

holtskinner committed:
Fix docstring formatting

dizcology commented on google/cloud/documentai_toolbox/wrappers/document.py:
```diff
@@ -147,24 +151,37 @@ def _get_batch_process_metadata(
         if operation.done:
             break
 
+        time.sleep(polling_interval)
```
Can we use the built-in `retry` here, instead of managing polling ourselves?

https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-documentai/google/cloud/documentai_v1/services/document_processor_service/client.py#L3532

holtskinner replied:
That is a much better way to implement, I thought based on the documentation that was only used for errors, not incomplete operations, but I can change it to use that.

holtskinner committed:
Change Operation Polling to use default api_core retry.

- Required significant refactoring to convert the operations_pb into a Google API Core Operation that can be polled.

holtskinner committed:
test: Fix unit tests for `_get_batch_process_metadata()`
